### PR TITLE
Overdue books

### LINKED ISSUE #6: As a librarian, I want to generate a list of overdue books
so that I can know which books are overdue and contact their borrowers if I have to

### PR DESCRIPTION
**Pull request by hoholyin**
closes #6

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -86,6 +96,9 @@ public void setGuiSettings(GuiSettings guiSettings) {
         userPrefs.setGuiSettings(guiSettings);
     }
 
+<<<<<<< HEAD
+
+=======
```
Comment (seeziyang) [inline src/main/java/seedu/address/model/ModelManager.java]: Merge conflict

**Review by seeziyang — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (hoholyin) [reply src/main/java/seedu/address/model/ModelManager.java]: resolved in latest commit

File: src/main/java/seedu/address/commons/util/DateUtil.java [hunk 2]
```diff
@@ -19,11 +19,11 @@ public static LocalDate getTodayDate() {
     }
 
     /**
-     * Get the date that is days number of days after starting date.
+     * Get the date that is number of days after starting date.
```
Comment (seeziyang) [inline src/main/java/seedu/address/commons/util/DateUtil.java:22]: Thanks for the English fix

File: src/main/java/seedu/address/model/Catalog.java [hunk 3]
```diff
@@ -113,6 +114,29 @@ public void removeBook(Book key) {
...
+        return FXCollections.observableArrayList(loanedBooks);
+    }
+
+    public ObservableList<BorrowerId> getOverdueBooksBorrowersId() {
+        List<BorrowerId> borrowersId = getOverdueBooks()
+                .stream()
+                .filter(book -> book.isCurrentlyLoanedOut()) // safety check to ensure loan is present
+                .map(book -> book.getLoan().get()) // get Loan object from each book
+                .map(loan -> loan.getBorrowerId()) // get BorrowerId from each Loan object
+                .collect(Collectors.toList()); // collect it to a list
```
Comment (seeziyang) [inline src/main/java/seedu/address/model/Catalog.java:136]: Great comments!

**Review by seeziyang — COMMENTED** [review]
LGTM! Thanks for the typo fixes!

**Review by seeziyang — APPROVED** [review]

File: src/main/java/seedu/address/model/BorrowerRecords.java [hunk 4]
```diff
@@ -31,22 +34,34 @@ public BorrowerRecords(ReadOnlyBorrowerRecords toBeCopied) {
         resetData(toBeCopied);
     }
 
+    /**
```
Comment (linyutinglyt) [inline src/main/java/seedu/address/model/BorrowerRecords.java:37]: Very neatly documented!

**Review by linyutinglyt — COMMENTED** [review]
LGTM!

HAND,
Yuting

**Pull request merged**